### PR TITLE
replace `run_cmd` with `run_shell_cmd` in custom easyblock for TensorFlow (`tensorflow.py`)

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _TensorFlow-2.13.0-foss-2023a.eb_ 
 * **SUCCESS** _TensorFlow-2.4.1-foss-2020b.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/7d3785bf6108f9f85e48fcd45ee59759 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _TensorFlow-2.11.0-foss-2022a-CUDA-11.7.0.eb_ (partial log available at https://gist.github.com/boegel/01d4c164e58ca52b6f58c274de2bc2a1)

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3306.joltik.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 535.154.05, Python 3.6.8
See https://gist.github.com/boegel/7677a729aeeb9b969303d73cc66e8d35 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
EB_BRANCH="5.0.x"
EB_ARGS="TensorFlow-2.13.0-foss-2022b.eb --installpath /tmp/$USER/pr3165"

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _TensorFlow-2.11.0-foss-2022a-CUDA-11.7.0.eb_ 
 * **SUCCESS** _h5py-3.1.0-fosscuda-2020b.eb_ 
 * **SUCCESS** _TensorFlow-2.5.0-fosscuda-2020b.eb_ 

Build succeeded for 3 out of 3 (2 easyconfigs in total)
bear-pg0208u15a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), 1 x NVIDIA NVIDIA A100-SXM4-40GB, 520.61.05, Python 3.6.8
See https://gist.github.com/branfosj/51075038a946d1c5b0cad2810dc650d3 for a full test report.

File: easybuild/easyblocks/t/tensorflow.py [hunk 1]
```diff
@@ -971,14 +971,15 @@ def test_step(self):
                 num_gpus_to_use = 0
             else:
                 # determine number of available GPUs via nvidia-smi command, fall back to just 1 GPU
-                # Note: Disable logging to also disable the error handling in run_cmd and do it explicitly below
-                (out, ec) = run_cmd("nvidia-smi --list-gpus", log_ok=False, log_all=False, regexp=False)
+                # Note: Disable logging to also disable the error handling in run_shell_cmd and do it explicitly below
```
Comment (boegel) [inline easybuild/easyblocks/t/tensorflow.py:974]:
```suggestion
                # Note: Disable checking exit code in run_shell_cmd, and do it explicitly below
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _TensorFlow-2.11.0-foss-2022a-CUDA-11.7.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3306.joltik.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 535.154.05, Python 3.6.8
See https://gist.github.com/boegel/e30343b04c8bd1833fcc865c94acb642 for a full test report.

**Pull request merged**